urbansim_templates/data/column_from_expression.py

```diff
@@ -0,0 +1,194 @@
+        Tags to associate with the template instance.
+    
+    autorun : bool, default True
+        Whether to run automatically when the template instance is registered with 
+        ModelManager.
+    
+    """
+    def __init__(self, 
+            column_name = None, 
+            table = None, 
+            expression = None, 
+            data_type = None, 
+            missing_values = None, 
+            cache = False, 
+            cache_scope = 'forever', 
+            name = None,
+            tags = [], 
+            autorun = True):
+        
+        # Template-specific params
+        self.column_name = column_name
+        self.table = table
+        self.expression = expression
+        self.data_type = data_type
+        self.missing_values = missing_values
+        self.cache = cache
+        self.cache_scope = cache_scope
+        
+        # Standard params
+        self.name = name
+        self.tags = tags
+        self.autorun = autorun
+        
+        # Automatic params
+        self.template = self.__class__.__name__
+        self.template_version = __version__
+    
+    
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Create an object instance from a saved dictionary representation.
+        
+        Parameters
+        ----------
+        d : dict
+        
+        Returns
+        -------
+        Table
+        
+        """
+        obj = cls(
+            column_name = d['column_name'],
+            table = d['table'],
+            expression = d['expression'],
+            data_type = d['data_type'],
+            missing_values = d['missing_values'],
+            cache = d['cache'],
+            cache_scope = d['cache_scope'],
+            name = d['name'],
+            tags = d['tags'],
+            autorun = d['autorun']
+        )
+        return obj
+    
+    
+    def to_dict(self):
+        """
+        Create a dictionary representation of the object.
+        
+        Returns
+        -------
+        dict
+        
+        """
+        d = {
+            'template': self.template,
+            'template_version': self.template_version,
+            'name': self.name,
+            'tags': self.tags,
+            'autorun': self.autorun,
+            'column_name': self.column_name,
+            'table': self.table,
+            'expression': self.expression,
+            'data_type': self.data_type,
+            'missing_values': self.missing_values,
+            'cache': self.cache,
+            'cache_scope': self.cache_scope,
+        }
+        return d
+    
+    
+    def run(self):
+        """
+        Run the template, registering a column of derived data with Orca.
+        
+        Requires values to be set for ``column_name``, ``table``, and ``expression``.
+        
+        Returns
+        -------
+        None
+        
+        """
+        if self.column_name is None:
+            raise ValueError("Please provide a column name")
```

Comment:
This line uses double quotes but there are single quotes in other parts of the file. In the future we can use pylint to ensure uniform code style.

Reply:
I use double quotes for natural-language messages and single quotes for strings that are more like data, i think. Not sure if this is a standard or just a habit i picked up, though! We can see what pylint wants.